lib/ansible/plugins/action/network_get.py

```diff
@@ -0,0 +1,132 @@
+# (c) 2015, Ansible Inc,
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import copy
+import os
+import time
+import re
+import q
+
+from ansible.module_utils._text import to_text
+from ansible.module_utils.connection import Connection
+from ansible.errors import AnsibleError
+from ansible.plugins.action import ActionBase
+from ansible.module_utils.six.moves.urllib.parse import urlsplit
+
+try:
+    from __main__ import display
+except ImportError:
+    from ansible.utils.display import Display
+    display = Display()
+
+
+class ActionModule(ActionBase):
+
+    def run(self, tmp=None, task_vars=None):
+        socket_path = None
+        play_context = copy.deepcopy(self._play_context)
+        play_context.network_os = self._get_network_os(task_vars)
+
+        result = super(ActionModule, self).run(task_vars=task_vars)
+
+        if play_context.connection != 'network_cli':
+            # It is supported only with network_cli
+            result['failed'] = True
+            result['msg'] = ('please use network_cli connection type for network_get module')
+            return result
+
+        try:
+            src = self._task.args.get('src')
+        except KeyError as exc:
+            return {'failed': True, 'msg': 'missing required argument: %s' % exc}
+
+        # Get destination file if specified
+        dest = self._task.args.get('dest')
+
+        if dest is None:
+            dest = self._get_default_dest(src)
+        else:
+            dest = self._handle_dest_path(dest)
+            q(dest)
+
+        # Get proto
+        proto = self._task.args.get('protocol')
+        if proto is None:
+            proto = 'scp'
+
+        sock_timeout = play_context.timeout
```

Comment:
Based on PR https://github.com/ansible/ansible/pull/39367 the timeout value need to be fetched from get_option
```
self.get_option('persistent_command_timeout')

```

Reply:
Sure I will change it once I this change is available in devel.